src/trans.c

```diff
@@ -2022,6 +2020,8 @@ Obj FuncON_KERNEL_ANTI_ACTION(Obj self, Obj ker, Obj f, Obj n){
     if(len>=deg){
       pttmp=ResizeInitTmpTrans(len);
       ptf2=ADDR_TRANS2(f);
+      out=NEW_PLIST(T_PLIST_CYC+IMMUTABLE, len);
+      SET_LEN_PLIST(out, len);
     
```

Comment:
I've permuted lines (2021,2023)(2022,2024) and the three other similar bits as you suggest, and it seems to fix most of the remaining problems that we can find - all tests pass.  But this relies on our (perhaps invalid) use of `FuncDegreeOfTransformation`, so I won't modify the pull request.